### PR TITLE
update blocklist and config loading

### PR DESCRIPTION
**Pull request by arcolife**
move url skip list to a config file loader

and add a common should skip function across indexers

also fixes client search in query.py to derive it from central init db func.

later: Open an issue for regex and sophisticated parsers

### CODE REVIEW

**Comment by tarekziade** [comment]
Looks promising! for the blocklist, I wonder if we can also add regexps. 

Thinking about catching all pages that have a known authentication pattern, like oauth2 dances etc,

**Comment by tarekziade** [comment]
Typical pattern for oauth:

```
https://<server>auth?response_type=<type>&client_id=<id>&redirect_uri=<url>
```

so maybe:

```
^(?=.*\bresponse_type\b)(?=.*\bclient_id\b).*$
```

**Comment by tarekziade** [comment]
what about ad servers? maybe we could use https://github.com/anudeepND/blacklist/blob/master/adservers.txt

**Comment by tarekziade** [comment]
also: `URL_SKIPLIST` should be added into the `/index` web API so we drop URLs there and that will include calls from the extension, not just the places db loader

**Comment by arcolife** [comment]
Right yeah I feel like we should discuss this. Seems like it needs a lot more thought into it as there are so many patterns and URLs. That regex will be unmanageable after a while (let alone searchable).

I'm thinking something that works browser's cache/cookies/url parsers or maybe pattern lists from adblockers.

**Comment by tarekziade** [comment]
> Right yeah I feel like we should discuss this. Seems like it needs a lot more thought into it as there are so many patterns and URLs. That regex will be unmanageable after a while (let alone searchable).
> 
> I'm thinking something that works browser's cache/cookies/url parsers or maybe pattern lists from adblockers.

sounds good let's do this as a follow up

**Review by tarekziade — APPROVED** [review]

**Comment by tarekziade** [comment]
I would drop `3.8` and `3.9` support unless you think there's a specific use case to run against those python versions

**Comment by arcolife** [comment]
will cover the rest here https://github.com/myinternets/places/issues/27

File: places/utils.py [hunk 1]
```diff
@@ -15,23 +15,22 @@
...
-    "127.0.0.1",
-    "0.0.0.0",
-)
 _QA = pipeline("question-answering", model="distilbert-base-cased-distilled-squad")
 
 
-def should_skip(url):
-    return urlparse(url).hostname in _SKIP
+def should_skip(url, cache=None):
+    url_hostname = urlparse(url).hostname
```
Comment (tarekziade) [inline places/utils.py:25]: you don't use that `url_hostname` (it's overwritten in the next for line)

Comment (arcolife) [reply places/utils.py:25]: nice catch.
```py
    if url_hostname in URL_SKIP_LIST:
        return True
 ```
 
 this should work

**Review by tarekziade — APPROVED** [review]

**Pull request merged**